File: Strings/Easy/LargestOddNumberInString.py
# 1903
class Solution:
    def largestOddNumber(self, num: str) -> str:
        
        n = len(num)
        last = n - 1
        flag = False
        for c in range(n - 1, -1, -1):
            if int(num[c]) % 2:
                flag = True
                last = c
                break
        return num[:last + 1] if flag else ""
        

File: GoLang/Strings/Easy/LargestOddNumberInString.go
// 1903
package easy

func largestOddNumber(num string) string {

    for i := len(num) - 1; i >= 0; i-- {
        digit := num[i] - '0'
        if digit % 2 == 1 {
            return num[:i + 1]
        }
    }
    return ""
}